kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/InCommitTimestampSuite.scala

```diff
@@ -0,0 +1,236 @@
+      val table = Table.forPath(engine, tablePath)
+      val txnBuilder = table.createTransactionBuilder(engine, testEngineInfo, CREATE_TABLE)
+
+      val txn1 = txnBuilder
+        .withSchema(engine, testSchema)
+        .build(engine)
+
+      txn1.commit(engine, emptyIterable())
+
+      val ver0Snapshot = table.getLatestSnapshot(engine).asInstanceOf[SnapshotImpl]
+      assertMetadataProp(ver0Snapshot, IN_COMMIT_TIMESTAMPS_ENABLED, false)
+      assert(!ver0Snapshot.getProtocol.getWriterFeatures.contains("inCommitTimestamp-preview"))
+      assert(getInCommitTimestamp(engine, table, 0).isEmpty)
+
+      val beforeCommitAttemptStartTime = System.currentTimeMillis
+      setTablePropAndVerify(
+        engine = engine,
+        tablePath = tablePath,
+        isNewTable = false,
+        key = IN_COMMIT_TIMESTAMPS_ENABLED,
+        value = "true",
+        expectedValue = true)
+
+      val ver1Snapshot = table.getLatestSnapshot(engine).asInstanceOf[SnapshotImpl]
+      assert(ver1Snapshot.getProtocol.getWriterFeatures.contains("inCommitTimestamp-preview"))
+      assert(
+        getInCommitTimestamp(engine, table, 1).get >= beforeCommitAttemptStartTime)
+    }
+  }
+
+  test("Enablement tracking properties should not be added if ICT is enabled on commit 0") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      setTablePropAndVerify(
+        engine = engine,
+        tablePath = tablePath,
+        key = IN_COMMIT_TIMESTAMPS_ENABLED,
+        value = "true",
+        expectedValue = true)
+
+      val ver0Snapshot =
+        Table.forPath(engine, tablePath).getLatestSnapshot(engine).asInstanceOf[SnapshotImpl]
+      assertHasNoMetadataProp(ver0Snapshot, IN_COMMIT_TIMESTAMP_ENABLEMENT_TIMESTAMP)
+      assertHasNoMetadataProp(ver0Snapshot, IN_COMMIT_TIMESTAMP_ENABLEMENT_VERSION)
+    }
+  }
+
+  test("Enablement tracking works when ICT is enabled post commit 0") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val table = TableImpl.forPath(engine, tablePath)
+      val txnBuilder = table.createTransactionBuilder(engine, testEngineInfo, CREATE_TABLE)
+
+      val txn = txnBuilder
+        .withSchema(engine, testSchema)
+        .build(engine)
+
+      txn.commit(engine, emptyIterable())
+
+      appendData(
+        engine,
+        tablePath,
+        schema = testSchema,
+        partCols = Seq.empty,
+        data = Seq(Map.empty[String, Literal] -> dataBatches1),
+        tableProperties = Map(IN_COMMIT_TIMESTAMPS_ENABLED.getKey -> "true"))
+
+      val ver1Snapshot = table.getLatestSnapshot(engine).asInstanceOf[SnapshotImpl]
+      assertMetadataProp(ver1Snapshot, IN_COMMIT_TIMESTAMPS_ENABLED, true)
+      assertMetadataProp(
+        ver1Snapshot,
+        IN_COMMIT_TIMESTAMP_ENABLEMENT_TIMESTAMP,
+        Optional.of(getInCommitTimestamp(engine, table, version = 1).get))
+      assertMetadataProp(
+        ver1Snapshot,
+        IN_COMMIT_TIMESTAMP_ENABLEMENT_VERSION, Optional.of(1L))
```

Comment:
```suggestion
      assertMetadataProp(
        ver1Snapshot,
        IN_COMMIT_TIMESTAMP_ENABLEMENT_VERSION,
        Optional.of(1L))
```

Reply:
same in other places: either keep all args in one line or each on its own separate line.